hooks.go

```diff
@@ -21,14 +21,38 @@ func (hooks LevelHooks) Add(hook Hook) {
 	}
 }
 
+// multiErr may contain multiple errors.
+type multiErr []error
+
+func (e multiErr) Error() string {
+	if e == nil || len(e) == 0 {
+		return ""
+	}
+	return e[0].Error()
+}
+
+func (e multiErr) Unwrap() error {
+	if e == nil || len(e) == 0 {
+		return nil
+	}
+	return e[0]
+}
+
 // Fire all the hooks for the passed level. Used by `entry.log` to fire
-// appropriate hooks for a log entry.
+// appropriate hooks for a log entry. In case of an error, the first error
+// encountered will be returned, but all hooks will fire.
 func (hooks LevelHooks) Fire(level Level, entry *Entry) error {
+	merr := make(multiErr, 0)
```

Comment:
append handles nil slices, so this isn't necessary: https://play.golang.org/p/9M1TC9lnkcQ